packages/mattermost-redux/src/reducers/entities/emojis.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {combineReducers} from 'redux';
+
+import {EmojiTypes, PostTypes, UserTypes} from 'mattermost-redux/action_types';
+import {EmojisState, CustomEmoji} from 'mattermost-redux/types/emojis';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {Post} from 'mattermost-redux/types/posts';
+import {IDMappedObjects} from 'mattermost-redux/types/utilities';
+
+export function customEmoji(state: IDMappedObjects<CustomEmoji> = {}, action: GenericAction): IDMappedObjects<CustomEmoji> {
+    switch (action.type) {
+    case EmojiTypes.RECEIVED_CUSTOM_EMOJI: {
+        const nextState = {...state};
+        nextState[action.data.id] = action.data;
+        return nextState;
+    }
+    case EmojiTypes.RECEIVED_CUSTOM_EMOJIS: {
+        const nextState = {...state};
```

Comment:
A bunch of files wrote this sort of thing as `const nextState: any = {...state};` for some reason. TypeScript didn't like that, so I removed the unnecessary type change everywhere.